Source/JavaScriptCore/wasm/WasmBBQJIT.cpp

```diff
@@ -7349,11 +7370,12 @@ class BBQJIT {
             m_jit.vectorExtractLaneInt64(TrustedImm32(0), left.asFPR(), m_scratchGPR);
             m_jit.vectorExtractLaneInt64(TrustedImm32(0), right.asFPR(), m_dataScratchGPR);
             m_jit.mul64(m_scratchGPR, m_dataScratchGPR, m_scratchGPR);
-            m_jit.vectorReplaceLane(SIMDLane::i64x2, TrustedImm32(0), m_scratchGPR, result.asFPR());
+            m_jit.vectorReplaceLane(SIMDLane::i64x2, TrustedImm32(0), m_scratchGPR, m_scratchFPR);
```

Comment:
Same as above, does this actually cause issues? Even though we write to `result.asFPR()` prior to the use of `left/right.asFPR()` below (which may alias), my expectation is that the lanes are distinct, and for each lane we obey the ordering that we write the result only after reading each input.

(For this one I'm mostly just curious, I still think the change looks good since it's less fragile and doesn't require additional scratches)

Reply:
Yes, `JSTests/wasm/v8/simd-i64x2-mul.js` is failing due to this.